app/models/maestrano/connector/rails/services/data_consolidator.rb

```diff
@@ -81,6 +81,7 @@ def consolidate_external_entities(external_entities, connec_entity_name)
         entity_id = @current_entity.class.id_from_external_entity_hash(entity)
         idmap = @current_entity.class.find_or_create_idmap(external_id: entity_id, organization_id: @organization.id, connec_entity: connec_entity_name.downcase)
 
+        next if @current_entity.class.immutable? && idmap.connec_id.present?
```

Comment:
Add a comment above and a blank line after.

Reply:
perhaps extract this to a public method on the Entity class (?)

Reply:
I think its better to extract it into private method in the same class. It doesn't belong in Entity class as it also queries idmap.